### ISSUE TITLE
Compilation fails on native silicon.

### ISSUE THREAD

**Issue by jdoege** [MumflrFumperdink/olcPGEMac#21]
Compiling on an M1 Macbook pro, the scrollWheelUpdate function (around line 4794) uses a ABI function called objc_msgSend_fpret. This is an x86_64 ABI not available on aarch64. On a guess, I tried replacing it with just objc_msgSend and the demo with the colorful boxes seemd to work. I have no idea if the scrollWheelUpdate function works. I then replaced that with this:

```
#if defined(__aarch64__) // change for native Apple silicon
            double deltaY = ((double (*)(id, SEL))objc_msgSend)(theEvent, deltaYSel);
#else
            double deltaY = ((double (*)(id, SEL))objc_msgSend_fpret)(theEvent, deltaYSel);
#endif
```

and things still seem to work. This was the only change needed to get it to compile. An unrelated warning remains, however:

olcPixelGameEngine.h:3290:14: 'SetDecalMode' overrides a member function but is not marked 'override'

**Comment by MumflrFumperdink** [comment]
Hi!

Thank you for suggesting this update. Unfortunately, I do not have an M1 MacBook, so I cannot test anything out in that regard, although I suspect that this replacement does not return an actual _double_ and more of a random bit of memory. If you can test objc_msgSend and it does not work with scrolling, can you try replacing it with objc_msgSend_fpret2 (don't know if it works, just seen it out there).

I just want to see if this change actually functions properly. A lot of the information on this stuff online only talks about PowerPC (PPC) macs vs intel (i386) macs, none talk about M1 macs, so I don't have a lot to go off of. Again, thank you for suggesting this!

**Comment by jdoege** [comment]
I found this: https://www.mikeash.com/pyblog/objc_msgsends-new-prototype.html

fpret is said to be an x86-64 thing. Maybe casting it will be sufficient?

If I ever get to using the scroll wheel, I guess I'll find out.

I am happy to test anything you do on an M1.

**Comment by MumflrFumperdink** [comment]
Yeah, it would be great if you could test this. Scrolling isn't essential, but it would be very nice if it worked on all Macs.
